Changed how choices are handled to fix #132.

Faxn opened the pull request:
* Changes `create.js` to never decrement numChoices. This prevents duplicate choices from happening.
* Changes `CreateView.post` in `views.py` to handle choices not being contiguous. (ie 'choice1', 'choice3')

#132

rupalkhilari commented:
Thanks @Faxn, I shall review and get back to you in a days time.

rupalkhilari commented on approval_polls/views.py:
```diff
@@ -412,21 +412,21 @@ def post(self, request, *args, **kwargs):
+                m = re.match("choice(\d+)", key)
+                if m:
+                    text = request.POST[key].strip()
                     choices.append(text)
```
The app allows blank options to be added as valid choices. Perhaps you should check if the `text` choice is empty before appending it to the `choices` (what line 421 did).

rupalkhilari commented on approval_polls/static/approval_polls/create.js:
```diff
@@ -117,10 +117,8 @@ $(function () {
-    console.log(e);
     var container = $(e.currentTarget).closest('.input-group');
     container.remove();
-    that.numChoiceFields--;
```
Now that we are not decrementing the choice number, i think we should also not include the number in the placeholder text for the textfield. See https://github.com/electology/approval_frame/blob/master/approval_polls/templates/approval_polls/create.html#L39 . i.e. I believe displaying `Choice` instead of `Choice <number>` would be sufficient for the user, and we use the choice numbers internally to keep track of them.

Faxn commented:
@rupalkhilari I believe I have addressed your concerns. Please look it over when you get a chance.

While working on the changes you suggested I found another bug I introduced. Since the keys of a dictionary have no specific order, The new method was randomly shuffling the order of the choices. I changed how the choices are stored and sort them based on id number so now the order in the create screen will always match what the poll winds up with.

rupalkhilari commented:
@Faxn , Good catch ! Thanks for addressing that too..Your changes look good, merging and closing this. 

Thanks!
